featuretools/tests/entityset_tests/test_es.py

```diff
@@ -929,7 +929,7 @@ def test_handles_datetime_mismatch():
 def test_dataframe_init(es):
     df = pd.DataFrame(
         {
-            "id": ["0", "1", "2"],
+            "id": [0, 1, 2],
```

Comment:
I think maybe we just want to set the LogicalType for ID on L943 (and remove L946) instead of relying on inference here? Maybe we should just set all the types regardless of the dataframe type. I'm not sure we need to rely on WW inference for anything in this test.